### DISCUSSION TITLE
How to add a new service in our eSchool?

### DISCUSSION THREAD

**Discussion by cseshohag** [OpenCodeFoundation/eSchool#339] [General]
How to add a new service in our eSchool?

**Comment by cseshohag** [comment]
https://medium.com/gdg-vit/clean-architecture-the-right-way-d83b81ecac6

**Reply by NitaiBanik** [reply]
Vhai ekhane clean architeure dekhte parben:
https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html

**Comment by ratanparai** [comment]
We need a guideline about how to add new service. Maybe when anyone is adding new service, he/she can document the approach. I will try my best to guide.

**Reply by mahedee** [reply]
Follow the Enrolling service in service folder.
1. Create a API project
2. Create a Domain Project
3. Create infrastructure project
4. Now modify API project following Enrolling.API project

**Reply by ratanparai** [reply]
Thanks. I will try to write an wiki entry

**Reply by ratanparai** [reply]
But on the other hand. We are not gonna add new service everyday and its better to learn by mistake for beginners. If someone follow a step by step guideline, then he/she will never learn it
